op-batcher/batcher/driver.go

```diff
@@ -516,21 +516,18 @@ func (l *BatchSubmitter) sendTransaction(ctx context.Context, txdata txData, que
 	if *candidate.To != l.RollupConfig.BatchInboxAddress {
 		return fmt.Errorf("candidate.To is not inbox")
 	}
-	if l.RollupConfig.UseInboxContract && l.inboxIsEOA == nil {
+	if l.inboxIsEOA == nil {
 		var code []byte
 		code, err = l.L1Client.CodeAt(ctx, *candidate.To, nil)
 		if err != nil {
 			return fmt.Errorf("CodeAt failed:%w", err)
 		}
 		isEOA := len(code) == 0
-		if !isEOA {
-			return fmt.Errorf("UseInboxContract is enabled but BatchInboxAddress is an EOA")
-		}
 		l.inboxIsEOA = &isEOA
 	}
 
-	// Don't set GasLimit when UseInboxContract is enabled so that later on `EstimateGas` will be called
-	if !l.RollupConfig.UseInboxContract {
+	// Don't set GasLimit when inbox is contract so that later on `EstimateGas` will be called
+	if !*l.inboxIsEOA {
```

Comment:
If the inbox is changed from an EOA to a smart contract, would we need to restart the batch, or is the system able to handle this transition smoothly?

Reply:
This is a great question!

I think we can not reliably cache the result because of the case [here](https://github.com/ethstorage/optimism/pull/59#issuecomment-2354596530), but maybe it's good enough if we just clear the cache when a transaction failure is spotted?

Reply:
@qzhodl Please take another look at the latest commit, which applies the above change.